consensus/src/consensus/utility.rs

```diff
@@ -24,97 +26,105 @@ impl Consensus {
     pub async fn create_coinbase_transaction(
         &self,
         block_num: u32,
-        transactions: &[SerialTransaction],
+        transactions: Vec<SerialTransaction>,
         program_vk_hash: Vec<u8>,
         new_birth_program_ids: Vec<Vec<u8>>,
         new_death_program_ids: Vec<Vec<u8>>,
         recipients: Vec<Address>,
     ) -> Result<TransactionResponse, ConsensusError> {
-        let mut rng = thread_rng();
-        let mut total_value_balance = crate::get_block_reward(block_num);
-
-        for transaction in transactions.iter() {
-            let tx_value_balance = transaction.value_balance;
+        let consensus = self.clone();
+        let (old_records, old_account_private_keys, new_records, memo) = task::spawn_blocking(move || {
```

Comment:
what's the rationale here? miner should already be in a owned blocking thread, and calling this from `block_on`.

Reply:
the general answer is provided later on, but more specifically, this is exactly a block containing blocking operations and calls to `thread_rng()`